_rules/inline-link-in-paragraph-has-distinguishable-style-on-hover-and-focus-548868.md

```diff
@@ -0,0 +1,317 @@
+---
+id: 548868
+name: Inline link has distinguishing style or content on hover and focus
+rule_type: atomic
+description: |
+  This rule checks that inline links are distinguishable from the surrounding text through a difference in style or content when the links are hovered or focused.
+accessibility_requirements:
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], for which all the following is true:
+
+- **link text**: the element has [visible][] [text nodes][text node] as [descendants][descendant] in the [flat tree][]; and
+- **non-link line text**: the element is [rendered on a line][] containing [visible][] [text nodes][text node] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][]; or
+- **different color**: the element's [foreground color][] and the [foreground color][] of the **non-link line text** elements have a [highest possible contrast][] that is at least 3.0:1, or the element's [background color][] and the [background color][] of the **non-link line text** elements have a [highest possible contrast][] that is at least 3.0:1.
+
+## Expectation
+
+For each test target, at least one of the following is true when it is [focused][] or [hovered][]:
+
+- **distinguishing style**: the element has a [distinguishing style][] not based on color alone from the **non-link line text** elements; or
```

Comment:
The `*-color` longhand properties cannot cover everything as you correctly point out. Missing out, at least:
- box-shadows
- background images that have only different colors
- color for fonts

That's why we kept the reference to color so "generic". It always meant to us "if you take away the color information you would still be able to distinguish the elements based on another style information".

I don't see how we would be able to convey this in a way that doesn't make this sentence really complex... Any suggestions?

Reply:
Yeah, I agree that we can't make it really good. 
I think that "not based on color alone" is OK-ish. As in I don't really see how else it could be interpreted and guess it isn't really ambiguous. So let's keep it…